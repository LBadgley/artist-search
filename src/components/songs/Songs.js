import React from 'react';
import Song from './Song';
import PropTypes from 'prop-types';

function Songs({ songs, artist }) {
  const songList = songs.map(song => {
    return (
      <li key={song.id}>
        <Song songTitle={song.title} artist={artist} />
      </li>
    );
  });

  return (
    <div>
      <ul>{songList}</ul>
    </div>
  );
}

Songs.propTypes = {
  songs: PropTypes.array.isRequired,
  artist: PropTypes.string
};

export default Songs;
