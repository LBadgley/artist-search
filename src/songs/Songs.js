import React from 'react';
import Song from './Song';
import PropTypes from 'prop-types';

function Songs({ songs }) {
  const songList = songs.map(song => {
    return (
      <li key={song.id}>
        <Song songName={song.title} songId={song.id} />
      </li>
    );
  });

  return (
    <ul>{songList}</ul>
  );
}

Songs.propTypes = {
  songs: PropTypes.array.isRequired
};

export default Songs;
