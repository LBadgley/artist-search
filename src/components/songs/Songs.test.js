import React from 'react';
import { shallow } from 'enzyme';
import Songs from './Songs';

describe('Songs test', () => {
  it('renders the Releases', () => {
    const songs = [
      {
        artist: 'artist1',
        songTitle: 'title1',
        id: 'billy'
      },
      {
        artist: 'artist2',
        songTitle: 'title3',
        id: 'strings'
      }
    ];
    const wrapper = shallow(<Songs songs={songs} />);
    expect(wrapper).toMatchSnapshot();
  });
});
