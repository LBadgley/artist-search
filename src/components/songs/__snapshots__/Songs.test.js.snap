// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs test renders the Releases 1`] = `
<div>
  <ul>
    <li
      key="billy"
    >
      <Song />
    </li>
    <li
      key="strings"
    >
      <Song />
    </li>
  </ul>
</div>
`;
